docs/how-to/update_similarity_indexes.md

```diff
@@ -0,0 +1,88 @@
+
+While Similarity Indexes can be used directly, they are usually used with [Views](../concepts/views.md), annotating arguments to filter methods. This technique lets db-ally automatically match user-provided arguments to the most similar value in the data source. You can see an example of using similarity indexes with views on the [Quickstart Part 2: Semantic Similarity](../quickstart/quickstart2.md) page.
+
+Similarity Indexes are designed to index all possible values (e.g., on disk or in a different data store). Consequently, when the data source undergoes changes, the Similarity Index must update to reflect these alterations. This guide will explain different ways to update Similarity Indexes.
+
+You can update the Similarity Index through Python code or via the db-ally CLI. The following sections explain how to update these indexes using both methods:
+
+* [Update Similarity Indexes via the CLI](#updating-similarity-indexes-via-the-cli)
+* [Update Similarity Indexes via Python Code](#updating-similarity-indexes-via-python-code)
+
+## Update Similarity Indexes via the CLI
+
+To update Similarity Indexes via the CLI, you can use the `dbally update-index` command. This command requires a path to what you wish to update. The path should follow this format: "path.to.module:ViewName.method_name.argument_name" where each part after the colon is optional. The more specific your target is, the fewer Similarity Indexes will be updated.
+
+For example, to update all Similarity Indexes in a module `my_module.views`, use this command:
+
+```bash
+dbally update-index my_module.views
+```
+
+To update all Similarity Indexes in a specific View, add the name of the View following the module path:
+
+```bash
+dbally update-index my_module.views:MyView
+```
+
+To update all Similarity Indexes within a specific method of a View, add the method's name after the View name:
+
+```bash
+dbally update-index my_module.views:MyView.method_name
+```
+
+Lastly, to update all Similarity Indexes in a particular argument of a method, add the argument name after the method name:
+
+```bash
+dbally update-index my_module.views:MyView.method_name.argument_name
+```
+
+For example, given the following view:
+
+## Update Similarity Indexes via Python Code
+### Update on a Single Similarity Index
+To manually update a Similarity Index, call the `update` method on the Similarity Index object. The `update` method will re-fetch all possible values from the data source and re-index them. Below is an example of how to manually update a Similarity Index:
+
+```python
+from db_ally import SimilarityIndex
+
+# Create a similarity index
+similarity_index = SimilarityIndex(fetcher=fetcher, store=store)
+
+# Update the similarity index
+await similarity_index.update()
+```
+
+### Detect Similarity Indexes in Views
```

Comment:
Maybe it would be beneficial to add extra method on collection to update all indexes used inside? Something like Collection.update_indexes() which aggregates all SimilarityIndexes and executes update on them

Reply:
Maybe. I viewed the CLI as the primary method of updating collections, and focused less on ways of doing this in code that can't be used by CLI, but you may be right. On the other hand this seems like a separate area, so how about we open a new ticket and do this in a new PR?